bindings/tflite/java/org/tensorflow/lite/Interpreter.java

```diff
@@ -0,0 +1,436 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.tensorflow.lite;
+
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * Main driver class for the IREE Java compatibility shim. Provides model
+ * creation and inference for IREE compatible TFLite models.
+ *
+ * <p>This shim aims to mimic the functionality of the Tensorflow Lite's
+ * Interpeter.java class, however, there are a few notable features IREE dosn't
+ * support:
+ *
+ * <ul>
+ *  <li> Delegates and the NNAPI
+ *  <li> Advanced interpreter options
+ *  <li> Interrupting or canceling inference before its complete
+ *  <li> Tensor method signatures
+ *  <li> Dynamic shapes
+ *  <li> String input/output isn't supported
+ * <ul>
+ *
+ * <p>In addition, this compatibility shim only accepts {@link java.nio.Buffer}
+ * type input. Consumers with scalar and array inputs can see conversion methods
+ * below.
+ *
+ * <p>Example using the Interpreter with a model with a single input/output Tensor:
+ *
+ * <pre>{@code
+ * ByteBuffer modelByteBuffer = ... load model here ....
+ * try (Interpreter interpreter = new Interpreter(modelByteBuffer)) {
+ *   interpreter.allocateTensors();
```

Comment:
This usage example wraps the entire process in a `try()`, which ties the `Interpreter` lifetime, buffer manipulation, and synchronous inference call together in a single scope. We could split this up into a few sections:

```
// Load model and initialize an Interpreter.

// Prepare inputs.

// Run inference.

// Cleanup.
```

The way this is written now is more concise though. WDYT?

(though I sort of realize that I'm reviewing TFLite's documentation here... https://www.tensorflow.org/lite/api_docs/java/org/tensorflow/lite/Interpreter)

Reply:
I've cleaned this up a bit. Let me know if this ls clearer.

Reply:
Yep! This usage example code LGTM. Thanks